### PR TITLE
docs: update description of useFragment hook

### PR DESCRIPTION
**Pull request by chaunceyau**
Think there's a misconception that the only use of fragments is to re-use and not have to redefine fields. This doesn't seem to be the primary benefit of fragments, i.e. if it were, it'd just push us in the direction of rest implementations where we're over-fetching data. The primary benefit, imo is the ability to define sub-selections of data dependencies to colocate and independently evolve those with your components.

### Checklist:

- [x] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [x] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Review by jerelmiller — APPROVED** [review]
This is a great change. Thanks for helping us make our docs better!

**Comment by jerelmiller** [comment]
@chaunceyau fair warning, this might take a little bit before we can merge. We've had some issues with our CLA bot that is currently in the process of getting fixed. Once that is back up and running, we'll be able to get this merged.

File: docs/source/api/react/hooks-experimental.mdx [hunk 1]
```diff
@@ -20,7 +20,7 @@ Beginning with version `3.7.0`, Apollo Client Web has preview support for the `u
 
 ### Using `useFragment_experimental`
 
-> A [GraphQL fragment](http://graphql.org/learn/queries/#fragments) is a piece of logic that can be shared between multiple queries and mutations. [See the API reference.](../../data/fragments)
+> A [GraphQL fragment](http://graphql.org/learn/queries/#fragments) defines a subselection of fields on a GraphQL type, that can help define component-level data dependencies or allow re-use between multiple queries and mutations. [See the API reference.](../../data/fragments)
```
Comment (alessbell) [inline docs/source/api/react/hooks-experimental.mdx:23]: Small tweak: our [fragment docs](https://www.apollographql.com/docs/react/data/fragments/) use the term "subset":

```suggestion
> A [GraphQL fragment](http://graphql.org/learn/queries/#fragments) defines a subset of fields on a GraphQL type that can be used to specify component-level data dependencies or allow re-use between multiple queries and mutations. [See the API reference.](../../data/fragments)
```

**Review by alessbell — CHANGES_REQUESTED** [review]
Left one small comment but once that's updated this will be a nice improvement. Thanks!

File: docs/source/api/react/hooks-experimental.mdx (hunk 1 above)
Comment (chaunceyau) [reply docs/source/api/react/hooks-experimental.mdx:23]: I love that, ty! Updated 👍

**Comment by chaunceyau** [comment]
> @chaunceyau fair warning, this might take a little bit before we can merge. We've had some issues with our CLA bot that is currently in the process of getting fixed. Once that is back up and running, we'll be able to get this merged.

No worries at all, completely understand! 🤞

**Review by alessbell — APPROVED** [review]
CLA bot is fixed now so I can get this merged, thanks!

**Pull request merged**